server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1321,6 +1332,87 @@ public Host maintain(final PrepareForMaintenanceCmd cmd) {
         }
     }
 
+    /**
+     * Declares host as Degraded. This method is used in critical situations; e.g. if it is not possible to start host, not even via out-of-band.
+     */
+    @Override
+    public Host declareHostAsDegraded(final DeclareHostAsDegradedCmd cmd) throws NoTransitionException {
+        Long hostId = cmd.getId();
+        HostVO host = _hostDao.findById(hostId);
+
+        if (host == null || host.getRemoved() != null) {
+            throw new InvalidParameterValueException(String.format("Host [id:%s] does not exist", host.getId()));
```

Comment:
Better to display hostname instead of id?

Reply:
Good idea, I have updated the code; however, it still necessary to log just the ID either if the host is null or the hostname is null/blank, assuming that the user has provided an ID that does not result in a valid Host.